xrpl/wallet/main.py

```diff
@@ -16,69 +17,131 @@ class Wallet:
+        master_address: Optional[str] = None,
+        seed: Optional[str] = None,
     ) -> None:
         """
         Generate a new Wallet.
 
         Args:
-            seed: The seed from which the public and private keys are derived.
-            sequence: The next sequence number for the account.
-            algorithm: The algorithm used to encode the keys. Inferred from the seed if
-                not included.
-        """
-        self.seed = seed
-        """
-        The core value that is used to derive all other information about
-        this wallet. MUST be kept secret!
+            public_key: The public key for the account.
+            private_key: The private key used for signing transactions for the account.
+            master_address: Include if a Wallet uses a Regular Key Pair. It must be
+                the master address of the account. The default is `None`.
+            seed: The seed used to derive the account keys. The default is `None`.
         """
-
-        pk, sk = derive_keypair(self.seed, algorithm=algorithm)
-        self.public_key = pk
+        self.public_key = public_key
         """
         The public key that is used to identify this wallet's signatures, as
         a hexadecimal string.
         """
 
-        self.private_key = sk
+        self.private_key = private_key
         """
         The private key that is used to create signatures, as a hexadecimal
         string. MUST be kept secret!
         """
 
-        self.classic_address = derive_classic_address(self.public_key)
-        """The address that publicly identifies this wallet, as a base58 string."""
+        self._classic_address = (
+            ensure_classic_address(master_address)
+            if master_address is not None
+            else derive_classic_address(self.public_key)
+        )
+        """Internal variable for classic_address. Use classic_address instead."""
 
-        self.sequence = sequence
+        self.seed = seed
         """
-        The next available sequence number to use for transactions from this
-        wallet.
-        Must be updated by the user. Increments on the ledger with every successful
-        transaction submission, and stays the same with every failed transaction
-        submission.
+        The core value that is used to derive all other information about
+        this wallet. MUST be kept secret!
         """
 
     @classmethod
-    def create(
-        cls: Type[Wallet], crypto_algorithm: CryptoAlgorithm = CryptoAlgorithm.ED25519
+    def generate(
+        cls: Type[Wallet], algorithm: CryptoAlgorithm = CryptoAlgorithm.ED25519
     ) -> Wallet:
         """
         Generates a new seed and Wallet.
 
         Args:
-            crypto_algorithm: The key-generation algorithm to use when generating the
-                seed. The default is Ed25519.
+            algorithm: The digital signature algorithm to generate an address for.
+                The default is Ed25519.
```

Comment:
I think the previous explanation is a bit more accurate (While the seed can be used to derive an address, the main point of this is to get a seed or public/private keys)